File: 3213-count-subarrays-where-max-element-appears-at-least-k-times/3213-count-subarrays-where-max-element-appears-at-least-k-times.c
long long countSubarrays (int* nums, int numsSize, int k) {
    int max = 0;
    for (int i = 0; i < numsSize; i++)
        if (nums [i] > max)
            max = nums [i];
    long long count = 0, l = 0, maxFound = 0;
    for (int r = 0; r < numsSize; ++r) {
        if (nums [r] == max) maxFound++;
        while (maxFound == k) {
            count += numsSize - r;
            if (nums [l] == max) maxFound--;
            l++;
        }
    }
    return count;
}

File: 3213-count-subarrays-where-max-element-appears-at-least-k-times/3213-count-subarrays-where-max-element-appears-at-least-k-times.java
class Solution {
    public long countSubarrays (int [] nums, int k) {
        long count = 0, ans = 0;
        int maxN = Arrays.stream (nums).max ().getAsInt ();
        int left = 0, n = nums.length;
        for (int right = 0; right < n; right++) {
            if (nums [right] == maxN) count++;
            while (count >= k) {
                ans += n - right;
                if (nums [left] == maxN) count--;
                left++;
            }
        }
        return ans;
    }
}